components/concept/engine/src/main/java/mozilla/components/concept/engine/Engine.kt

```diff
@@ -51,6 +51,15 @@ interface Engine {
         override fun hashCode() = types
     }
 
+    /**
+     * Starts up this engine and makes sure all required initialization logic is executed. The
+     * details are specific to individual implementations, but the following must be true:
+     *
+     * - The engine must be operational after this method was called successfully
+     * - Calling this method on an engine that is already started has no effect
+     */
+    fun start() = Unit
+
```

Comment:
I wonder if we should override this in`GeckoEngine` implementations and do a thread/process check? 🤔 - But maybe that's more something for the creation of the object.

Reply:
Yeah, I agree it would be have to when we create the instance, but not sure. `Gecko` will already crash if we're on the wrong thread. Wdyt?

Reply:
Oh yeah, right. It only didn't for the process, I think. But that may be too late anyways here.